src/Compilers/CSharp/Test/Symbol/Symbols/ExtensionMethodTests.cs

```diff
@@ -4079,5 +4079,29 @@ void Validator(ModuleSymbol module)
 
             CompileAndVerify(source, validator: Validator, options: TestOptions.ReleaseDll);
         }
+
+        [Fact]
+        [WorkItem(65020, "https://github.com/dotnet/roslyn/issues/65020")]
+        public void ReduceExtensionsMethodOnReceiverTypeSystemVoid()
+        {
+            var source =
+@"static class C
+{
+    static void M(this object x)
+    {
+    }
+}";
+            var compilation = CreateCompilationWithMscorlib40AndSystemCore(source);
+            compilation.VerifyDiagnostics();
+
+            var extensionMethod = compilation.GlobalNamespace.GetMember<NamedTypeSymbol>("C").GetMember<MethodSymbol>("M");
+            Assert.True(extensionMethod.IsExtensionMethod);
+
+            var systemVoidType = compilation.GetSpecialType(SpecialType.System_Void);
+            Assert.Equal(SpecialType.System_Void, systemVoidType.SpecialType);
+
+            var reduced = extensionMethod.ReduceExtensionMethod(systemVoidType, null!);
```

Comment:
>null! [](http://example.com/codeflow?start=80&length=5)

Why not pass compilation?

Reply:
Added an assertion with compilation passed in